recursion/fibonacci.py

```diff
@@ -23,3 +23,20 @@
 # Ex:
 #   fibonacci(10) # return 55
 #   fibonacci(12) # return 144
+def fibonacci(number: int) -> int:
+    if number < 0:
+        return -1
+
+    memo = [None] * (number + 1)
+    memo[0] = 0
+    memo[1] = 1
+
+    def fibonacciHelper(number: int) -> int:
+        if memo[number] is not None:
```

Comment:
I told you that if memo[sum] is None, then memo[sum] is False otherwise it is True. You don’t need to check if it is not None. Just:

    if memo[sum]:
        <code block>